File: .old/PAT2021/A1108.py
n = int(input())
array = input().split()
sum_of_number = 0
num_of_number = 0.0

def is_num(a):
    flag = 0
    count = 0
    for i in range(len(a)):
        if '0' <= a[i] <= '9' or (flag == 0 and a[i] == '.') or (i == 0 and a[i] == '-'):
            if a[i] == '.':
                flag = 1
            elif flag == 1:
                count += 1
                if count > 2: return False
        else:
            return False 
    temp = float(a)
    if -1000.0 <=temp <= 1000.0:
        return True
    else:
        return False
    
for i in range(n):
    if(is_num(array[i])):
        num_of_number += 1
        sum_of_number += float(array[i])
    else:
        print(f'ERROR: {array[i]} is not a legal number')
    
if num_of_number == 0:
    print('The average of 0 numbers is Undefined')
elif num_of_number == 1:
    print('The average of 1 number is %.2f' %(sum_of_number))
else:
    print('The average of %d numbers is %.2f' %(num_of_number, sum_of_number / num_of_number))

File: .old/Aclass/A1108.cpp
#include <iostream>
#include <cstring>
using namespace std;
bool isNum(string num) {
    int cntde = 0;
    bool dot = false;
    for(int i = 0; i < num.size(); i++) {
        if(i == 0 && num[i] == '-') continue;
        else if(dot == false && num[i] == '.') dot = true;
        else if(num[i] <= '9' && num[i] >= '0') {
            if(dot == true) cntde++;
            if(cntde > 2) return false;
            continue;
        }
        else return false;
    }
    double temp = stof(num);
    if(temp <= 1000 && temp >= -1000) return true;
    else return false;
}
int main() {
    int n, cnt = 0;
    double sum = 0.0;
    cin >> n;
    for(int i = 0; i < n; i++) {
        string num;
        cin >> num;
        if(!isNum(num)) cout << "ERROR: " << num << " is not a legal number" << endl;
        else {
            double temp = stof(num);
            sum += temp;
            cnt++;
        }
    }
    if(cnt == 0) cout << "The average of 0 numbers is Undefined" << endl;
    else if(cnt == 1) printf("The average of 1 number is %.02f\n", sum);
    else printf("The average of %d numbers is %.02f\n", cnt, sum / cnt);
    return 0;
}